packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts

```diff
@@ -1017,6 +1040,13 @@ export class UserPool extends UserPoolBase {
     });
     this.emailConfiguration = emailConfiguration;
 
+    if (
+      props.featurePlan && props.featurePlan !== FeaturePlan.LITE &&
+      props.advancedSecurityMode && props.advancedSecurityMode !== AdvancedSecurityMode.OFF
+    ) {
+      throw new Error('you cannot enable Advanced Security Mode when feature plan is Essentials or higher.');
+    }
```

Comment:
If the advanced security mode is enabled with Essentials or Plus feature plan, CloudFormation will fail with following error:
> Resource handler returned message: "The following features need to be disabled for the ESSENTIALS pricing tier configured: Threat Protection (Service: CognitoIdenti
tyProvider, Status Code: 400, Request ID: xx)"

We cannot validate `advancedSecurityMode` is off when `featurePlan` is not specified (defaults to Essentials) because existing user pools are set to Lite feature plan for backward compatibility and CDK cannot determine what the actual feature plan is.